File: generator/src/test/resources/kotlin/expected/RamlObjectTypesTest/test-generated-classes-for-object-hierarchy.output4.kt
package io.test

import kotlin.Any
import kotlin.Boolean
import kotlin.Int
import kotlin.String

public class Test3(
  `value`: String,
  value2: String,
  public val value3: String,
) : Empty(value, value2) {
  public fun copy(
    `value`: String? = null,
    value2: String? = null,
    value3: String? = null,
  ): Test3 = Test3(value ?: this.value, value2 ?: this.value2, value3 ?: this.value3)

  override fun hashCode(): Int {
    var result = 31 * super.hashCode()
    result = 31 * result + value3.hashCode()
    return result
  }

  override fun equals(other: Any?): Boolean {
    if (this === other) return true
    if (javaClass != other?.javaClass) return false

    other as Test3

    if (value != other.value) return false
    if (value2 != other.value2) return false
    if (value3 != other.value3) return false

    return true
  }

  override fun toString(): String = """
  |Test3(value='$value',
  | value2='$value2',
  | value3='$value3')
  """.trimMargin()
}

File: generator/src/test/resources/swift/expected/RamlObjectTypesTest/test-generated-classes-for-object-hierarchy.output4.swift
import Sunday

public class Test3 : Empty {

  public var value3: String
  public override var debugDescription: String {
    return DescriptionBuilder(Test3.self)
        .add(value, named: "value")
        .add(value2, named: "value2")
        .add(value3, named: "value3")
        .build()
  }

  public init(
    value: String,
    value2: String,
    value3: String
  ) {
    self.value3 = value3
    super.init(value: value, value2: value2)
  }

  public required init(from decoder: Decoder) throws {
    let container = try decoder.container(keyedBy: CodingKeys.self)
    self.value3 = try container.decode(String.self, forKey: .value3)
    try super.init(from: decoder)
  }

  public override func encode(to encoder: Encoder) throws {
    try super.encode(to: encoder)
    var container = encoder.container(keyedBy: CodingKeys.self)
    try container.encode(self.value3, forKey: .value3)
  }

  public override func withValue(value: String) -> Test3 {
    return Test3(value: value, value2: value2, value3: value3)
  }

  public override func withValue2(value2: String) -> Test3 {
    return Test3(value: value, value2: value2, value3: value3)
  }

  public func withValue3(value3: String) -> Test3 {
    return Test3(value: value, value2: value2, value3: value3)
  }

  fileprivate enum CodingKeys : String, CodingKey {

    case value3 = "value3"

  }

}
